articles/WideStrings.md

```diff
@@ -48,87 +48,83 @@ For more information, the following links provide introductory material:
+ROS 2 will instead aim to be compatible with DDS-XTYPES 1.2 and use 16bit storage for wide characters.
+Generated code for ROS 2 messages will automatically handle the conversion when a message is serialized or deserialized.
 
-Pros:
+### Bounded wide strings
 
-* No conversions necessary for user-to-ROS 2 API.
+Embedded systems accepting strings may use message definitions that restrict the maximum size of a string.
+These are referred to as bounded strings.
+Their purpose is to restrinct the amount of memory used, so the bounds must be specified as units of memory.
+If a `string` field is bounded then the size is given in bytes.
+Similarly the size of a bounded `wstring` is to be specified in words.
+It is the responsibility of whoever populates a bounded `string` or `wstring` to make sure it contains whole code points only.
+Partial code points are indistinuguishable from unknown encodings, so a bounded string whose last code point is incomplete will be published without error.
 
-Cons:
+## Runtime impact of wide string
 
-* The user must always tell ROS 2 what encoding the data is in.
-* The encoding needs to be transmitted on the wire to the other side.
-* It is not clear how ROS 2 will transmit the encoding.
-  Fixed list?
-  Which encodings go into the list?
+Dealing with wide strings puts more strain on the software of a system, both in terms of speed and code size.
+UTF-8 and UTF-16 are both variable width encodings, meaning a code point can take 1 (UTF-8 only), 2, or 4 bytes to represent.
+It may take multiple code points to represent a single user perceived character.
+One of the goals of ROS 2 is to support microcontrollers that are contrained by both code size and processor speed.
+Some wide string operations like splitting a string on a user perceived character may not be possible on these devices.
 
-While a user defined encoding looks somewhat attractive at first, the downsides to it make it difficult to determine exactly which encodings are in use.
-The downsides also mean that any program that wants to parse ROS 2 messages (or bag files) may potentially have to deal with many different encodings.
-Additionally, Unicode (and in particular, UTF-8) are very commonplace now, while other encodings are becoming more esoteric.
-Even earlier Unicode (UTF-16 and UTF-32) encodings are essentially deprecated in favor of UTF-8.
-For these reasons, the current design of the ROS 2 wide string is to use a UTF-8 encoding for all strings.
+However, whole string equality checking is the same whether using wide strings or not.
+Further splitting a UTF-8 string on an ASCII character is identical to splitting an ASCII character on an ASCII string.
+If a microcontroller must do other types of string manipluation, then a `string` type should still be used.
+Code in the subscriber callback should stop processing a message when it encounters a byte greater than 127 in a `string` field.
```

Comment:
Can you clarify this recommendation?

Reply:
It's trying to justify using the `string` type on microcontrollers that need to do string manipulation by saying it is cheap to filter non-ascii values when using UTF-8. I'll expand this paragraph to clear it up.

Reply:
Tried to make this clearer in 9c4e7b5ab5316d27d62d4e083b035ec27201cd73